### ISSUE TITLE
slick grid 5.4.2. type   type WeakKey was not declared, sortable.js fails to load

### ISSUE THREAD

**Issue by b-giavotto** [6pac/SlickGrid#909]
the source file in npm package does not declare WeakKey in slick.core.d.ts
plus, even declaring this as any, during execution a statements like the following : 
let grid = new SlickGrid(el, data, columns, options);
fail because it does not load sortable.js

Here is my package.json
{
  "name": "myapp-client-bundle",
  "version": "1.0.0",
  "description": "This is client-side scripts bundle for MyApp",
  "private": true,
  "scripts": {
    "build": "webpack --config webpack.config.js --mode=development",
    "build:prod": "webpack --mode=production"
  },
  "devDependencies": {
    "@types/jquery": "^3.5.16",
    "@types/jqueryui": "^1.12.17",
    "ts-loader": "^9.4.3",
    "typescript": "^5.1.3",
    "webpack": "^5.86.0",
    "webpack-cli": "^5.1.4"
  },
  "dependencies": {
    "jquery": "^3.7.0",
    "jquery-ui-touch-punch": "^0.2.3",
    "jqueryui": "^1.11.1",
    "linq-to-typescript": "^11.0.0",
    "slickgrid": "^5.4.2",
    "sortablejs": "^1.15.0"
  }
}

**Comment by ghiscoding** [comment]
duplicate of #907, you have to import SortableJS yourself and add it to the `window` object yourself. I did not find a way to properly import it directly in the project because we need to support ESM, CJS and IIFE. Also we use esbuild to package all these 3 targets and everything I tried so far had a few disadvantages that were mentioned in #907. 

So with that being a duplicate and because of the disadvantage, I will repeat the same conclusion. The easiest is to import SortableJS yourself and add it to the `window` object. I also did say in that same discussion that a note was added to the [v5.0 Migration](https://github.com/6pac/SlickGrid/wiki/Major-version-5.0-%E2%80%90-ES6-ESM-and-TypeScript-Support#es6--esm-cjs-esm) about this same window object. 

in summary, I'll close this as "work as intended" unless you find a better way that works for all 3 build targets, if so please contribute a PR

**Comment by b-giavotto** [comment]
What about the WeakKey type?

**Comment by ghiscoding** [comment]
We don't use WeakKey, neither need, anywhere in SlickGrid, so I don't understand why you ask about it. You're not providing enough details to understand the problem, the only thing I understood is about SortableJS which I wrote about

**Comment by b-giavotto** [comment]
> We don't use WeakKey, neither need, anywhere in SlickGrid, so I don't understand why you ask about it. You're not providing enough details to understand the problem, the only thing I understood is about SortableJS which I wrote about

Sorry, i've discovered the issue later. Please ignore the previous message. WeakKey appear in slick.core.d.ts but it is really contained in some of the typescript library definition (es5 or so.). I got this message because of a misconfigured visual studio project, that try to compile typescript source with invalid settings in tsconfig.json. After disabling the compilation task in MSBuild and relying on my webpack confiuguration the compilation goes well.
